lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -0,0 +1,299 @@
+    except:
+        raise Exception("{}: // {}".format("e", policy_json_string))
+
+        # returns nothing, so we have to just assume it didn't throw
+        ret['changed'] = True
+
+    ret['changes_needed'] = changes_needed
+    ret['have_invalid_entries'] = have_invalid_entries
+    if dry_run:
+        # true if changes > 0
+        ret['changed'] = (not len(changes_needed) == 0)
+
+    return ret
+
+def assert_policy_shape(policy):
+    '''Since the policy seems a little, uh, fragile, make sure we know approximately what we're looking at.'''
+    errors = []
+    if policy['Version'] != "2012-10-17":
+        errors.append('Unknown version/date ({}) of policy. Things are probably different than we assumed they were.'.format(policy['Version']))
+
+    found_statement_type = {}
+    for statement in policy['Statement']:
+        for label,sidlabel in statement_label.items():
+            if statement['Sid'] == sidlabel:
+                found_statement_type[label] = True
+
+    for statementtype in statement_label.keys():
+        if not found_statement_type.get(statementtype):
+            errors.append('Policy is missing {}.'.format(statementtype))
+
+    if len(errors):
+        raise Exception('Problems asserting policy shape. Cowardly refusing to modify it: {}'.format(' '.join(errors)))
+    return None
+
+def main():
+    argument_spec = ansible.module_utils.ec2.ec2_argument_spec()
+    argument_spec.update(dict(
+            mode = dict(choices=['grant', 'deny'], default='grant'),
+            key_alias = dict(required=False, type='str'),
+            key_arn = dict(required=False, type='str'),
+            role_name = dict(required=False, type='str'),
+            role_arn = dict(required=False, type='str'),
+            grant_types = dict(required=False, type='list'),
+            clean_invalid_entries = dict(type='bool', default=True),
+        )
+    )
+
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec=argument_spec,
+        required_one_of=[['key_alias', 'key_arn'], ['role_name', 'role_arn']],
+        required_if=[['mode', 'grant', ['grant_types']]]
+    )
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    result = {}
+    mode = module.params['mode']
+
+
+    try:
+        region, ec2_url, aws_connect_kwargs = ansible.module_utils.ec2.get_aws_connection_info(module, boto3=True)
+        kms = ansible.module_utils.ec2.boto3_conn(module, conn_type='client', resource='kms', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        iam = ansible.module_utils.ec2.boto3_conn(module, conn_type='client', resource='iam', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.NoCredentialsError as e:
+        module.fail_json(msg='cannot connect to AWS', exception=traceback.format_exc(e))
+
+
+    if mode == 'grant' or mode == 'deny':
```

Comment:
You don't need this conditional, since Ansible enforces that these are the only choices.